Dockerfiles/pcluster-amazonlinux-2/Dockerfile

```diff
@@ -109,18 +109,23 @@ RUN curl -sOL https://efa-installer.amazonaws.com/aws-efa-installer-${EFA_INSTAL
     && cd .. \
     && rm -rf aws-efa-installer-${EFA_INSTALLER_VERSION}.tar.gz aws-efa-installer
 
-# Bootstrap spack compiler installation
-RUN mkdir -p /bootstrap && \
-    cd /bootstrap && \
-    git clone https://github.com/spack/spack spack \
-    && export SPACK_ROOT=/bootstrap/spack \
-    && . spack/share/spack/setup-env.sh \
-    && curl -sOL https://raw.githubusercontent.com/spack/spack-configs/main/AWS/parallelcluster/postinstall.sh \
-    && /bin/bash postinstall.sh -fg -nointel \
-    && spack clean -a \
-    && cd /bootstrap/spack \
-    && find . -type f -maxdepth 1 -delete \
-    && rm -rf bin lib share var /root/.spack
+# Bootstrap spack compiler installation into the eventual installation tree
+# Defined in spack/share/spack/gitlab/cloud_pipelines/configs/config.yaml
+ARG SPACK_ROOT="/bootstrap-compilers/spack"
+#  commit after https://github.com/spack/spack/pull/34821
+ARG SPACK_COMMIT=f27d012
+ARG TARGETARCH
+COPY Dockerfiles/pcluster-amazonlinux-2/packages-${TARGETARCH}.yaml /root/.spack/packages.yaml
+RUN mkdir -p $(dirname "${SPACK_ROOT}") \
+    && git clone https://github.com/spack/spack "${SPACK_ROOT}" \
+    && pushd "${SPACK_ROOT}" && git checkout ${SPACK_COMMIT} && popd \
+    && cp "${SPACK_ROOT}/share/spack/gitlab/cloud_pipelines/configs/config.yaml" "${SPACK_ROOT}/etc/spack/config.yaml" \
```

Comment:
I wonder if there's a way to do this, such that a future inconsistency due to a change to `config.yaml` would be noticed immediately :thinking:

Reply:
I see what you mean. If we keep a copy of that `config.yaml` around and compare it in the pipeline, what to do we do when it has changed? I don't think the pipeline should throw and error, it should rather silently abort.

Reply:
We can fail/warn in the `before_script` with a message saying that `config.yaml` changed and the image needs an update.

Reply:
OK, I'll keep it around.